subsys/ipc/rpmsg_multi_instance/rpmsg_multi_instance.c

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ipc/rpmsg_multi_instance.h>
+#include <zephyr.h>
+#include <device.h>
+#include <logging/log.h>
+
+#include <drivers/ipm.h>
+#include <openamp/open_amp.h>
+#include <metal/cache.h>
+
+LOG_MODULE_REGISTER(rpmsg_multi_instance, CONFIG_RPMSG_MULTI_INSTANCE_LOG_LEVEL);
+
+#define SHM_DEVICE_NAME          "sram0.shm"
+#define IPM_MSG_ID               CONFIG_IPM_MSG_ID
+
+#if CONFIG_CACHE_MANAGEMENT
+#define RPMSG_MI_FLUSH(x)      metal_cache_flush(&x, sizeof(x))
+#define RPMSG_MI_INVALIDATE(x) metal_cache_invalidate(&x, sizeof(x))
+#else
+#define RPMSG_MI_FLUSH(x)      do { } while (0)
+#define RPMSG_MI_INVALIDATE(x) do { } while (0)
+#endif
```

Comment:
This is not needed, you only have to use `sys_cache_data_range()`

Reply:
Fixed.